### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by bindi-nagda**
This adds tests for targets `apply`, `multigrid`, `time_stepping`, `time_stepping_thn `and `frm`.

### CODE REVIEW

**Comment by bindi-nagda** [comment]
The tests that were added check that the solver outputs the expected error_norms for `apply`, `multigrid`, `time_stepping`, `time_stepping_thn`. For target `frm`, the tests simply check that both the newtonian and complex fluid simulation can run for a few steps. It doesn't look at the output to check for number of iterations to converge or relative residuals. If we need to check for other items for the frm solve, I can certainly add them.

@abarret We need to add the tests for convec_op and physical_bcs. This PR can be merged into main in the meantime. I would be happy to add the tests for physical_bc and convec_op either by pushing directly to their branch or making a new PR later, whichever is more appropriate.

**Review by abarret — CHANGES_REQUESTED** [review]

**Comment by abarret** [comment]
> number of iterations to converge or relative residuals
These things are sensitive to finite precision order of operations and are not reproducible.

> I would be happy to add the tests for physical_bc and convec_op either by pushing directly to their branch or making a new PR later, whichever is more appropriate.

We can make a new PR for those. I would rather get those merged in sooner.

**Comment by bindi-nagda** [comment]
Going to update this PR to resolve merge conflicts, and add additional unit tests for convection operator and physical boundary conditions.

**Comment by abarret** [comment]
> Going to update this PR to resolve merge conflicts, and add unit tests for convection operator and physical boundary conditions.

Can you just resolve the merge conflicts, and add new unit tests to a separate PR? I would like to go ahead and get this merged.

**Comment by bindi-nagda** [comment]
@abarret After rebasing with main, I ran into some issues with unit tests failing for targets `apply `and `multigrid`. Using gdb, I found seg faults occurring when calling the method `CacheBcCoefData` in `VCTwoFluidStaggeredStokesOperator`. I fixed these issues by creating boundary condition objects and passing them to stokes_op. This seemed to work with all unit tests now passing, but perhaps there's a better way to do this?

Otherwise this is ready to merge..

**Comment by abarret** [comment]
Yeah. I think the better approach is to make the physical bc helper optional in the operator. I'll push that change in the morning and merge.

File: include/multiphase/VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.h [hunk 1]
```diff
@@ -268,10 +268,10 @@ class VCTwoFluidStaggeredStokesBoxRelaxationFACOperator : public IBTK::FACPrecon
     double d_nu_s = std::numeric_limits<double>::quiet_NaN();
 
     SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM>> d_hierarchy; // Reference patch hierarchy
-    std::vector<SAMRAI::solv::RobinBcCoefStrategy<NDIM>*> d_un_bc_coefs;
-    std::vector<SAMRAI::solv::RobinBcCoefStrategy<NDIM>*> d_us_bc_coefs;
     std::unique_ptr<SAMRAI::solv::RobinBcCoefStrategy<NDIM>> d_default_un_bc_coef, d_default_us_bc_coef,
         d_default_P_bc_coef, d_default_thn_bc_coef;
+    std::vector<SAMRAI::solv::RobinBcCoefStrategy<NDIM>*> d_un_bc_coefs;
+    std::vector<SAMRAI::solv::RobinBcCoefStrategy<NDIM>*> d_us_bc_coefs;
```
Comment (abarret) [inline include/multiphase/VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.h:274]: Note that this is one reason to pay attention to compiler warnings in #70. These needed to be initialized in a different order than specified in the header.
